code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -359,24 +360,52 @@
 		P.fire()
 		return P
 
+
 /mob/living/simple_animal/hostile/proc/CanSmashTurfs(turf/T)
 	return iswallturf(T) || ismineralturf(T)
 
-/mob/living/simple_animal/hostile/proc/DestroySurroundings()
+
+/mob/living/simple_animal/hostile/proc/DestroyPathToTarget()
+	if(environment_smash)
+		EscapeConfinement()
+
+		var/dir_to_target_cardinal = get_dir(targets_from, get_cardinal_step_towards(targets_from, target))
+		var/dir_to_target = get_dir(targets_from, target)
+		var/dir_list = list()
+		if(dir_to_target in GLOB.diagonals) //it's diagonal, so we need two directions to hit
+			for(var/direction in GLOB.cardinals)
+				if(direction & dir_to_target)
+					dir_list += direction
+		else
+			dir_list += dir_to_target_cardinal
+		var/turf/T
+		for(var/direction in dir_list) //now we hit all of the directions we got in this fashion, since it's the only directions we should actually need
+			T = get_step(targets_from, direction)
+			if(!T.Adjacent(targets_from))
+				continue //we can't hit it, so continue to the next
+			if(CanSmashTurfs(T))
+				T.attack_animal(src)
+			for(var/a in T)
+				var/atom/A = a
+				if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
+					A.attack_animal(src) //we neither break nor return here, so that it hits both directions at once instead of possibly being inefficient. this is still better than hitting every possible turf at once, since it'll be aimed at the target.
+
+
+mob/living/simple_animal/hostile/proc/DestroySurroundings() // for use with megafauna destroying everything around them
 	if(environment_smash)
 		EscapeConfinement()
 		for(var/dir in GLOB.cardinals)
 			var/turf/T = get_step(targets_from, dir)
-			if(CanSmashTurfs(T))
```

Comment:
should probably keep this, but change this code to be this... and make it a proc, shared between this and DestroyPathToTarget(), to reduce code duplication and make changing it easier.
````
/mob/living/simple_animal/hostile/proc/DestroyObjectsInDirection(direction)
 	var/turf/T = get_step(targets_from, direction)
 	if(T.Adjacent(targets_from))
 		if(CanSmashTurfs(T))
 			T.attack_animal(src)
 		for(var/a in T)
 			var/atom/A = a
 			if(!A.Adjacent(targets_from))
 				continue
 			if(is_type_in_typecache(A, environment_target_typecache) && !A.IsObscured())
 				A.attack_animal(src)
````

Reply:
can you put that code in, ive lost p much all will to keep going with this bug
been very mentally frustrating, but overall a rewarding learning experience i guess
its like you think youve done it and realize that you need to reformat everything youve written, add a couple procs, remove those procs, go back to the source
maybe im not fit to be a programmer

Reply:
ill fix it tomorrow